File: manual/data/python/src/function_jsonschema_validate.py
from jsonschema import validate, ValidationError
import json

class JSONValidator:
    def validate_json(self, data_str: str, schema_str: str) -> str:
        try:
            data = json.loads(data_str)
            schema = json.loads(schema_str)
            validate(instance=data, schema=schema)
            return "True"
        except ValidationError:
            return "False"
        except Exception:
            return "Validation failed: Invalid JSON format"

File: manual/code/generate_strategies/IR(CoT)/java/java/function_jsonschema_validate.java
package com.example;

import org.everit.json.schema.Schema;
import org.everit.json.schema.ValidationException;
import org.everit.json.schema.loader.SchemaLoader;
import org.json.JSONObject;
import org.json.JSONException;

public class JSONValidator {
    public String validate_json(String data_str, String schema_str) {
        class JSONValidator {
            public String validateJson(String dataStr, String schemaStr) {
                try {
                    JSONObject dataJson = new JSONObject(dataStr);
                    JSONObject schemaJson = new JSONObject(schemaStr);
                    Schema schema = SchemaLoader.load(schemaJson);
                    schema.validate(dataJson);
                    return "True";
                } catch (ValidationException e) {
                    return "False";
                } catch (JSONException e) {
                    return "Validation failed: Invalid JSON format";
                }
            }
        }
        return new JSONValidator().validateJson(data_str, schema_str);
    }
}